### DISCUSSION TITLE
Does extractVersion work with updateType checks?

### DISCUSSION THREAD

**Discussion by samithShetty** [renovatebot/renovate#10701] [Request Help]
This is related to my mutable tag issue discussed [here](https://github.com/renovatebot/renovate/discussions/10665), although this discussion is about a completely different approach.

To quickly recap, my private repos have mutable tags that resolve to be the latest version of that major release i.e. if the latest release of version 4 is `v4.8.9`, then there is a `v4` tag that resolves to be `v4.8.9`, and when `v4.9.0` comes out, `v4` will be updated to resolve to that instead. I am trying to get renovate to use these, particularly when going to the next major release, e.g. `v3.5.1` --> `v4`

I have already implemented a solution with extractVersion that works in a limited capacity and restricts me to only major releases, but am now trying to implement a sort of hybrid approach that extracts according to whether the current version is already in mutable-tag form or not, and will keep it in that same form. I also want to put in an additional condition that if it's a major update, perform the extract to make it mutable regardless of what format it's currently in. Here's what I have right now

```js
        { //custom terraform-modules regex manager
            fileMatch: ['.*\.tf'],
            matchStrings: ['source *= (\"|\')git::ssh:\\/\\/git@bitbucket\\.int\\.<company>\\.com\\/tf\\/(?<depName>.*?).git\\?ref=(?<currentValue>v(?<major>\\d+)(?:\\.(?<minor>\\d+))?(?:\\.(?<patch>\\d+))?)(\"|\')'],
            extractVersionTemplate: "{{#if minor}}{{#if isMajor}}(?<version>v\\d+){{else}}(?<version>.*){{/if}}{{else}}(?<version>v\\d+){{/if}}",
            lookupNameTemplate: "https://bitbucket.int.<company>.com/tf/{{{depName}}}.git",
            datasourceTemplate: "git-tags"
        }
```

From my testing of how the extractVersionTemplate behaves, the isMajor statement ALWAYS evaluates to false, even for major updates. When I try to accomplish the same thing with a separate packageRule for major updates, the rule goes through (I know since the PRBodyNote shows on major PRs), but extractVersion doesn't seem to get applied.
```js
packageRules: [
        {
            updateTypes: ["major"],
            extractVersion: "(?<version>v\\d+)",
            prBodyNotes: [":warning: :warning: This is a major update :warning: :warning:"]
        }
]
```
Does extractVersion/extractVersionTemplate not work with update type checks? is there any way around it?

**Comment by viceice** [comment]
I don't think so but maybe this can help you

https://docs.renovatebot.com/configuration-options/#matchcurrentversion
